core/src/main/scala/nl/vroste/zio/kinesis/client/zionative/Consumer.scala

```diff
@@ -322,30 +322,30 @@ object Consumer {
                                            ZStream.fail(e)
                                        case Right(EndOfShard(childShards @ _)) =>
                                          ZStream.fromZIO(
-                                           ZIO.logDebug(
+                                           ZIO.logInfo(
                                              s"Found end of shard for ${shardId}. " +
                                                s"Child shards are ${childShards.map(_.shardId).mkString(", ")}"
                                            ) *>
                                              checkpointer.markEndOfShard() *>
                                              leaseCoordinator.childShardsDetected(childShards)
                                          ) *> ZStream.empty
                                      }
-                                     .mapChunksZIO { chunk => // mapM is slow
-                                       chunk
+                                     .mapChunksZIO {
+                                       _
                                          .mapZIO(record => toRecords(shardId, record))
                                          .map(_.flatten)
-                                         .tap { records =>
```

Comment:
The bug shows up if we try to consume a shard without any remaining data in it (which mostly can happen with very low volume streams after a scaling operation), but with a previous record.

In this case the previous record will be used to set a maxSequenceNumber using `checkpointer.setMaxSequenceNumber(extendedSequenceNumber)`, but filtered out before it hits the downstream consumer (and checkpointer#stage).
When the shard ends, instead of committing a SHARD_END checkpoint, we just release the shard due to https://github.com/svroonland/zio-kinesis/blob/master/core/src/main/scala/nl/vroste/zio/kinesis/client/zionative/leasecoordinator/DefaultCheckpointer.scala#L68-L69